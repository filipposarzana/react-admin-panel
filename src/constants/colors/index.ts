export const colors = {
  gray050: 'rgb(241,241,241)',
  gray100: 'rgb(200,200,200)',
  gray300: 'rgb(118,118,118)',
  gray600: 'rgb(64,64,64)',
  gray900: 'rgb(46,46,46)',
  red900: 'rgb(225, 61, 6)',
  transparent: 'rgb(255, 255, 255, 0)',
  white: 'rgb(255, 255, 255)',
}

export type ColorName = keyof typeof colors
