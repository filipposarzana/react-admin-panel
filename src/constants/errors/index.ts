export const maximumAllowedRetries = 2

export const maximumAllowedRetriesError = 'Maximum allowed retries exceeded'
