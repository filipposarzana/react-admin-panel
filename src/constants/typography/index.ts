type TypographyFlavour = {
  fontSize: 40 | 34 | 26 | 20 | 16.5 | 15 | 12
  fontWeight: 400 | 500 | 700
  lineHeight: 50 | 40 | 36 | 30 | 25 | 19 | 16
}

export type TypographyKind = 'bodyL' | 'bodyM' | 'bodyS' | 'headingL' | 'headingM' | 'headingS'

export const typographyFlavours: Record<TypographyKind, TypographyFlavour> = {
  bodyL: {
    fontSize: 16.5,
    fontWeight: 700,
    lineHeight: 25,
  },
  bodyM: {
    fontSize: 15,
    fontWeight: 500,
    lineHeight: 19,
  },
  bodyS: {
    fontSize: 12,
    fontWeight: 400,
    lineHeight: 16,
  },
  headingL: {
    fontSize: 40,
    fontWeight: 700,
    lineHeight: 50,
  },
  headingM: {
    fontSize: 34,
    fontWeight: 700,
    lineHeight: 40,
  },
  headingS: {
    fontSize: 26,
    fontWeight: 500,
    lineHeight: 30,
  },
}
