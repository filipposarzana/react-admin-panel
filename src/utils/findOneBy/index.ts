import { User, users } from '~/db'

type Key = Extract<keyof User, 'name' | 'uuid'>

const findOneBy = <K extends Key>(prop: K) => (value: User[K]) =>
  users.find((user) => user[prop].toLowerCase() === value.toLowerCase())

export const findOneByName = findOneBy('name')

export const findOneByUuid = findOneBy('uuid')
