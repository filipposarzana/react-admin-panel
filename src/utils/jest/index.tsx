import { render } from '@testing-library/react'
import { renderHook } from '@testing-library/react-hooks'
import { ReactElement } from 'react'
import { createRenderer } from 'react-test-renderer/shallow'
import { StyleSheetManager } from 'styled-components'

export { act, fireEvent, waitFor } from '@testing-library/react'

export const renderer = {
  deep: (node: ReactElement) => {
    const r = render(<StyleSheetManager disableVendorPrefixes>{node}</StyleSheetManager>)

    return {
      ...r,
      output: r.container.firstChild,
    }
  },
  hook: <T extends unknown, R extends unknown>(hookFunction: (props: T) => R) => {
    const r = renderHook(hookFunction)

    return r
  },
  shallow: (node: ReactElement) => {
    const r = createRenderer()

    r.render(node)

    return {
      instance: r.getMountedInstance(),
      output: r.getRenderOutput(),
      unmount: r.unmount,
    }
  },
}
