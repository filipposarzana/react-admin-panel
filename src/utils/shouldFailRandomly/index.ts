export const shouldFailRandomly = () => Math.round(Math.random()) === 0
