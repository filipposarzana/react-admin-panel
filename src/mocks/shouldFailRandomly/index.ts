import { mocked } from 'ts-jest/utils'
import { shouldFailRandomly } from '~/utils/shouldFailRandomly'

export const mockedShouldFailRandomly = mocked(shouldFailRandomly)
