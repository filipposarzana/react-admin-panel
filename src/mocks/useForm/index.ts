import { mocked } from 'ts-jest/utils'
import { FormState, useForm, UseFormReturn } from 'react-hook-form'

type Values = Record<string, unknown>

const defaultFormState = {
  isSubmitted: false,
  isSubmitting: false,
  isValid: false,
}

type GetMockedFormParam = {
  formState?: Partial<FormState<Values>>
  values?: Values
}

export const getMockedForm = ({ formState, values = {} }: GetMockedFormParam = {}) =>
  (({
    formState: {
      ...defaultFormState,
      ...formState,
    },
    handleSubmit: jest.fn((cb: (data: Values) => void) => () => cb(values)),
    setValue: jest.fn(),
  } as unknown) as UseFormReturn<Values>)

export const useMockedForm = mocked(useForm)
