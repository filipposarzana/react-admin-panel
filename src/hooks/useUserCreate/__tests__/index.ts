import { maximumAllowedRetriesError } from '~/constants/errors'
import { mockedShouldFailRandomly } from '~/mocks/shouldFailRandomly'
import { renderer } from '~/utils/jest'
import { useUserCreate } from '..'

jest.mock('~/db')

jest.mock('~/utils/shouldFailRandomly')

jest.mock('uuid', () => ({
  v4: () => '0267eff2-b1ca-331c-be5b-bb5aa22a6e5b',
}))

afterEach(jest.clearAllMocks)

describe('useUserCreate', () => {
  it('returns a list of users with new user', () => {
    const { result } = renderer.hook(() => useUserCreate())

    expect(result.current({ friends: [], name: 'Added' })).toMatchSnapshot()
  })

  it('throws error after exceeding maximum failures', () => {
    mockedShouldFailRandomly.mockReturnValue(true)

    const { result } = renderer.hook(() => useUserCreate())

    expect(() => result.current({ friends: [], name: 'Added' })).toThrowError(maximumAllowedRetriesError)
  })
})
