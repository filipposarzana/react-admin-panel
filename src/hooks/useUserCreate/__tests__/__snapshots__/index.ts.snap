// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`useUserCreate returns a list of users with new user 1`] = `
Array [
  Object {
    "friends": Array [
      Object {
        "name": "A Friend",
        "uuid": "acc82963-61c3-37ec-b334-e8f6bf6b7f57",
      },
    ],
    "name": "A User",
    "uuid": "6bb0295f-8665-30a1-af76-6d1236b88971",
  },
  Object {
    "friends": Array [],
    "name": "A User Friend",
    "uuid": "acc82963-61c3-37ec-b334-e8f6bf6b7f57",
  },
  Object {
    "friends": Array [],
    "name": "A Friend of a Friend",
    "uuid": "34e65f4b-188c-37e3-a73c-4e0f244df9eb",
  },
  Object {
    "friends": Array [],
    "name": "Added",
    "uuid": "0267eff2-b1ca-331c-be5b-bb5aa22a6e5b",
  },
]
`;
